sinks/api/endpoint_test.go

```diff
@@ -0,0 +1,801 @@
+		}
+		res, err := req.make()
+		assert.Nil(t, err, fmt.Sprintf("unexpected erro %s", err))
+		assert.Equal(t, sinkCase.status, res.StatusCode, fmt.Sprintf("%s: expected status code %d got %d", sinkCase.desc, sinkCase.status, res.StatusCode))
+	}
+
+}
+
+func TestUpdateSink(t *testing.T) {
+	service := newService(map[string]string{token: email})
+	server := newServer(service)
+	defer server.Close()
+
+	sk, err := service.CreateSink(context.Background(), token, sink)
+	require.Nil(t, err, fmt.Sprintf("unexpected error: %s", err))
+
+	data := toJSON(updateSinkReq{
+		Name:        sk.Name.String(),
+		Description: sk.Description,
+		Backend:     sk.Backend,
+		Config:      sk.Config,
+		Tags:        sk.Tags,
+	})
+
+	cases := map[string]struct {
+		req         string
+		id          string
+		contentType string
+		auth        string
+		status      int
+	}{
+		"update existing sink": {
+			req:         data,
+			id:          sk.ID,
+			contentType: contentType,
+			auth:        token,
+			status:      http.StatusOK,
+		},
+		"update sink with a empty json request": {
+			req:         "{}",
+			id:          sk.ID,
+			contentType: contentType,
+			auth:        token,
+			status:      http.StatusBadRequest,
+		},
+		"update sink with a invalid id": {
+			req:         data,
+			id:          "invalid",
+			contentType: contentType,
+			auth:        token,
+			status:      http.StatusNotFound,
+		},
+		"update non-existing sink": {
+			req:         data,
+			id:          wrongID.String(),
+			contentType: contentType,
+			auth:        token,
+			status:      http.StatusNotFound,
+		},
+		"update sink with invalid user token": {
+			req:         data,
+			id:          sk.ID,
+			contentType: contentType,
+			auth:        "invalid",
+			status:      http.StatusUnauthorized,
+		},
+		"update sink with empty user token": {
+			req:         data,
+			id:          sk.ID,
+			contentType: contentType,
+			auth:        "",
+			status:      http.StatusUnauthorized,
+		},
+		"update sink with invalid content type": {
+			req:         data,
+			id:          sk.ID,
+			contentType: "invalid",
+			auth:        token,
+			status:      http.StatusUnsupportedMediaType,
+		},
+		"update sink without content type": {
+			req:         data,
+			id:          sk.ID,
+			contentType: "",
+			auth:        token,
+			status:      http.StatusUnsupportedMediaType,
+		},
+		"update sink with a empty request": {
+			req:         "",
+			id:          sk.ID,
+			contentType: contentType,
+			auth:        token,
+			status:      http.StatusBadRequest,
+		},
+		"update sink with a invalid data format": {
+			req:         invalidJson,
+			id:          sk.ID,
+			contentType: contentType,
+			auth:        token,
+			status:      http.StatusBadRequest,
+		},
+		"update sink with different owner": {
+			req:         invalidJson,
+			id:          sk.ID,
+			contentType: contentType,
+			auth:        token,
+			status:      http.StatusBadRequest,
+		},
```

Comment:
let's add a test to make sure you cannot update the read only fields: backend, status, error